.github/actions/setup-dfx/action.yml

```diff
@@ -0,0 +1,32 @@
+name: 'Setup DFX'
+description: Setup DFX
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/cache@v2
+      with:
+        path: |
+          ~/dfx
+          ~/.cache/dfinity
+        key: dfx-cache-0.11.1-2
+    # This step hangs on Github actions on Darwin for some reason, that
+    # is why we run this only on Linux for now
+    - name: Install DFX
+      shell: bash
+      env:
+        DFX_VERSION: 0.11.1
+      run: |
+        if [ -f ~/dfx/dfx ]
+        then
+          echo "DFX restored from cache"
+          ~/dfx/dfx --version
```

Comment:
nit: move this out of the "true" branch so that it also shows the version when installing without the cache